compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -1577,6 +1577,14 @@ pub struct ConstVid<'tcx> {
     pub phantom: PhantomData<&'tcx ()>,
 }
 
+/// An **effect** **v**ariable **ID**.
```

Comment:
Why are effects being represented with a separate vid?

There's a weird dichotomy happening here where effects are kinda treated like const and kinda not, and it's not exactly clear why. Shouldn't we be able to store whether a const vid comes from an effect parameter in the const vid table so it can participate it fallback later?

Reply:
I kinda just followed how int and float type variables also have their own vids. We'd probably want to ensure that we know all variables instantiated for an effect param are actually for an effect param, and having a separate vid helped by catching things like relating a normal const var with an effect var which should never happen.

Reply:
Yeah, maybe that's better. This definitely needs documentation though -- that's a design decision that (afaict? maybe i'm just out of the loop) isn't written anywhere, and it should be probably 😺

Reply:
added a comment